extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -474,7 +474,8 @@ private List<String> getUserDefinedRuntimeFilters(OpenApiConfig openApiConfig, I
 
     private List<String> getUserDefinedFilters(OpenApiConfig openApiConfig, IndexView index, OpenApiFilter.RunStage stage) {
         List<String> userDefinedFilters = new ArrayList<>();
-        Collection<AnnotationInstance> annotations = index.getAnnotations(DotName.createSimple(OpenApiFilter.class.getName()));
+        List<AnnotationInstance> annotations = index.getAnnotations(DotName.createSimple(OpenApiFilter.class.getName()))
+                .stream().distinct().collect(Collectors.toList());
```

Comment:
In which case could you end up with duplicates?

Reply:
I am not sure, I could not find out why, but there are.

Reply:
@Ladicek do you agree that we would need an example of this behavior to better understand what's going on?

Reply:
Presence of duplicate `AnnotationInstance`s sounds like a bug somewhere (in Jandex or whoever constructs the `IndexView` if it's a custom thing).

It may also be caused by usage of `CompositeIndex` that contains the same class multiple times. This is actually the most likely scenario, because Quarkus uses `CompositeIndex`es heavily, but the fix here would _not_ actually work (because `ClassInfo`'s equality is identity, so the duplicate `AnnotationInstance`s would still be considered distinct).